src/wallet/wallet.cpp

```diff
@@ -4785,165 +4032,118 @@ bool CWallet::Lock()
 bool CWallet::Unlock(const CKeyingMaterial& vMasterKeyIn, bool accept_no_keys)
 {
     {
-        LOCK(cs_KeyStore);
-        if (!SetCrypted())
-            return false;
-
-        bool keyPass = mapCryptedKeys.empty(); // Always pass when there are no encrypted keys
-        bool keyFail = false;
-        CryptedKeyMap::const_iterator mi = mapCryptedKeys.begin();
-        for (; mi != mapCryptedKeys.end(); ++mi)
-        {
-            const CPubKey &vchPubKey = (*mi).second.first;
-            const std::vector<unsigned char> &vchCryptedSecret = (*mi).second.second;
-            CKey key;
-            if (!DecryptKey(vMasterKeyIn, vchCryptedSecret, vchPubKey, key))
-            {
-                keyFail = true;
-                break;
+        LOCK(cs_wallet);
+        for (const auto& spk_man_pair : m_spk_managers) {
+            if (!spk_man_pair.second->Unlock(vMasterKeyIn, accept_no_keys)) {
+                return false;
             }
-            keyPass = true;
-            if (fDecryptionThoroughlyChecked)
```

Comment:
CWallet::fDecryptionThoroughlyChecked variable isn't actually used anymore, only assigned to. It should probably either be used or removed. Maybe it's not necessary anymore since keyman has the same variable.

Reply:
Removed it